### PR TITLE
Add node model

### PR DESCRIPTION
**Pull request by kpcyrd**
I've added a mock NodeStore_nodeForAddr method which returns a new Object which is a bit closer to the admin api.

I don't know how to actually call this method, though.

The address gets generated from the public key later.

Feel free to commit on this branch.

### CODE REVIEW

File: src/main/java/berlin/meshnet/cjdns/model/Node.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package berlin.meshnet.cjdns.model;
+
+public class Node {
+
+    public final String publicKey;
+
+    public final int linkCount;
```
Comment (benhylau) [inline src/main/java/berlin/meshnet/cjdns/model/Node.java]: What is `linkCount`?

**Comment by benhylau** [comment]
How can the Java layer do stuff like `getAddressFromPublicKey(publicKey)` or `encryptWithPrivateKey(arbitraryMessage, nonce)` through the native service?

File: src/main/java/berlin/meshnet/cjdns/model/Node.java (hunk 1 above)
Comment (kpcyrd) [inline src/main/java/berlin/meshnet/cjdns/model/Node.java]: The number of peers, one of the values returned by `nodeByAddr`

**Comment by kpcyrd** [comment]
`getAddressFromPublicKey` is simply `double_sha(publicKey)`, see https://github.com/kpcyrd/yrd/blob/master/cjdns.py#L180

I'm not sure if `encryptWithPrivateKey(arbitraryMessage, nonce)` is possible.

**Comment by benhylau** [comment]
Actually, I don't even need `encryptWithPrivateKey(arbitraryMessage, nonce)`. I just need to sign stuff (i.e. the credentials bundle) with the private key during peering, so the other party knows the credentials came with the authenticated operator of the node. As long as the Java layer has access to the private and public keys, this all can be implemented by integrating [NaCl](https://github.com/neilalexander/jnacl).

**Comment by kpcyrd** [comment]
You can read the private key from the config, but I'm not sure if it should be used for anything except cjdns.

**Comment by benhylau** [comment]
@kpcyrd See [Step 4 of this](https://github.com/benhylau/wrbt). Can't think of another way to verify other than signing with the cjdns private key.

**Comment by kpcyrd** [comment]
I was busy and couldn't comment on this, yet. Sorry for that, I'll try to contribute in the next view days.

**Comment by benhylau** [comment]
@kpcyrd I will work on this branch when I get to the Node part after the Credentials. Thanks!

**Comment by benhylau** [comment]
Rebased and created branch in this repo so it's easier to work with. Closing this PR, k?

See https://github.com/BerlinMeshnet/cjdns-android/pull/34

**Pull request closed without merging**